src/MerchantFrontend/src/app/core/state/state.service.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from '@angular/core';
+import { Authority } from '../models/authority.model';
+import { Merchant } from '../models/merchant.model';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StateService {
+
+  private _authority: Authority|null = null;
```

Comment:
Ich kenne States üblicherweise mit Observables. Genauer gesagt `Subject`s, damit auch auf Änderungen reagiert werden können.

As far as I know states are commonly done with `Subject`s to capture value changes. For example when a user changes to null on logout.

Reply:
yes, you are correct. I changed it. I wanted to avoid complexity, but in the long run it's better with observables.